elasticdl/python/worker/ps_client.py

```diff
@@ -92,3 +100,102 @@ def push_dense_parameters(self, parameters, ps_id, version):
         for p in parameters:
             serialize_ndarray(p.values, model.dense_parameters[p.name])
         self.ps_stubs[ps_id].push_model(model)
+
+    def push_gradients(
+        self, grads, edl_grads, learning_rate, model_versions,
+    ):
+        """
+        Push gradients to PS. There two kinds of gradients:
+         - gradients of normal layers
```

Comment:
Dense gradients of normal layers.

Reply:
Will fix it in next PRs

Reply:
Normal embedding layers have sparse gradients, while normal non-embedding layers have dense gradients.